### PR TITLE
fix(engineconn): fix the java option memory unit bug

### PR DESCRIPTION
**Pull request by jackxu2011**
### What is the purpose of the change

When the exchangis submit job, with JAVA_ENGINE_REQUEST_MEMORY `1024MB`,  the ec startup failed  

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-engineconn/linkis-engineconn-plugin-core/src/main/scala/org/apache/linkis/manager/engineplugin/common/util/NodeResourceUtils.scala [hunk 1]
```diff
@@ -35,18 +40,35 @@ object NodeResourceUtils {
...
+  def formatJavaOptionMemory(memory: String, defaultUnit: String): String = {
+    assert(StringUtils.isNotBlank(memory))
+    val memoryTmp = memory.toUpperCase(Locale.getDefault)
+    if (
+        memoryTmp.endsWith(JAVA_MEMORY_UNIT_G) || memoryTmp.endsWith(
+          JAVA_MEMORY_UNIT_M
+        ) || memoryTmp.endsWith(JAVA_MEMORY_UNIT_K)
+    ) {
+      memoryTmp
+    } else if (memoryTmp.endsWith(JAVA_MEMORY_UNIT_B)) {
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn/linkis-engineconn-plugin-core/src/main/scala/org/apache/linkis/manager/engineplugin/common/util/NodeResourceUtils.scala:61]: Xmx not support  unit MB,
If the value unit set by the user  not within the range（G/M/K/B）, whether to consider print  warn log and set it as the default memory 1g

Comment (jackxu2011) [reply linkis-computation-governance/linkis-engineconn/linkis-engineconn-plugin-core/src/main/scala/org/apache/linkis/manager/engineplugin/common/util/NodeResourceUtils.scala:61]: This is a compatiable with exchangis, if use GB, MB, KB, B,  just drop it.  is this ok?

Comment (casionone) [reply linkis-computation-governance/linkis-engineconn/linkis-engineconn-plugin-core/src/main/scala/org/apache/linkis/manager/engineplugin/common/util/NodeResourceUtils.scala:61]: This is okay. If the user accidentally enters the wrong unit such as GK, do we need to consider using the default value or add some warn logs

Comment (jackxu2011) [reply linkis-computation-governance/linkis-engineconn/linkis-engineconn-plugin-core/src/main/scala/org/apache/linkis/manager/engineplugin/common/util/NodeResourceUtils.scala:61]: in this case,  the check should use regex to check?   if the unit wit `GK`, the ec will startup failed.

Comment (casionone) [reply linkis-computation-governance/linkis-engineconn/linkis-engineconn-plugin-core/src/main/scala/org/apache/linkis/manager/engineplugin/common/util/NodeResourceUtils.scala:61]: > in this case, the check should use regex to check? if the unit wit `GK`, the ec will startup failed.

I think it's ok

**Review by casionone — APPROVED** [review]

**Pull request merged**